### PR TITLE
update dev testing instructions

### PR DESCRIPTION
**Pull request by ggilmore**

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -15,8 +15,25 @@
       - Make sure to include the sha256 digest for each image, which [ensures that each image pull is immutable](https://renovatebot.com/docs/docker/#digest-pinning). Use `docker inspect --format='{{index .RepoDigests 0}}' $IMAGE` to get the digest.
 
 - Open a PR and wait for buildkite to pass and for your changes to be approved, then merge and check out master.
-- Test what is currently checked in to master by [installing](docs/install.md) Sourcegraph on fresh cluster.
-- Create a git tag and push it to the repository:
+- Test what is currently checked in to master by [installing](docs/install.md) Sourcegraph on a fresh cluster:
+  - Provision a new cluster:
+    1.  Create a cluster unique name that is identifiable to you (e.g `ggilmore-test`) in the [Sourcegraph Auxiliary GCP Project](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-server&organizationId=1006954638239). You can create pool that has `3` nodes, each with `8` vCPUs and `30` GB memory (for a total of `24` vCPUs and `90` GB memory).
```
Comment (beyang) [inline README.dev.md]: Can we script this with something like https://github.com/sourcegraph/infrastructure/blob/master/sh/create-cluster.sh?

File: README.dev.md [hunk 2]
```diff
@@ -15,8 +15,25 @@
...
+        - See this screenshot, but note that the UI could have changed: ![](https://imgur.com/RuCyGX2.png)
+    1.  It’ll take a few minutes for it to be provisioned, you’ll see a green checkmark when it is done.
+    1.  Click on the `connect` button next to your cluster, it’ll give you a command to copy+paste in your terminal.
```
Comment (beyang) [inline README.dev.md]: https://github.com/sourcegraph/infrastructure/blob/master/sh/activate-cluster.sh should do the same thing.

File: README.dev.md [hunk 3]
```diff
@@ -15,8 +15,25 @@
...
+    1.  Run the command in your terminal. Once it finishes, run `kubectl config current-context`. It should tell you that it’s set up to talk to the cluster that you just created.
+    1.  In order to give yourself permissions to create roles on the cluster, run: `kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $YOUR_NAME@sourcegraph.com`
```
Comment (attfarhan) [inline README.dev.md]: Should there be a step here for port forwarding?:
```suggestion
    1.  In order to give yourself permissions to create roles on the cluster, run: `kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $YOUR_NAME@sourcegraph.com`
    1. Make the frontend port accessible at localhost:30080 by running `kubectl port-forward $(kubectl get pod -l app=sourcegraph-frontend -o template --template="{{(index .items 0).metadata.name}}") 30080:3080`
```

File: README.dev.md [hunk 4]
```diff
@@ -15,8 +15,25 @@
...
+    - Check to see that the latest `master` is working on a fresh cluster
+      1. Deploy the latest `master` to your new cluster
+         - You'll need to create a GCP Storage Class named `sourcegraph` with the same `zone` that you created your cluster in (see ["Configure a storage class"](./docs/configure.md#Configure-a-storage-class))
+      1. Use the instructions in [configure.md](./docs/configure.md) to add a repository, test that code intelligence is working on it, and do a couple searches
+    - Check the upgrade path from the previous release to the laster `master`
+      1. Tear down the cluster that you created above
+      1. Checkout the commit that contains the configuration for the previous release (e.g. the commit has `2.11.x` images if you're currently trying to release `2.12.x`, etc.), and use the instructions above to create a new cluster, deploy the older commit to it, and do the same smoke tests with the older version
+      1. Checkout the latest `master`, deploy the newer images to the same cluster (without tearing it down in between), and check to see the smoke test passes after the upgrade process
```
Comment (attfarhan) [inline README.dev.md]: This is done by `./kubectl-apply-all` IIRC. I think it's good to explicitly mention this.
```suggestion
      1. Checkout the latest `master`, deploy the newer images to the same cluster (without tearing it down in between), and check to see the smoke test passes after the upgrade process by running `./kubectl-apply-all.sh`
```

File: README.dev.md (hunk 1 above)
Comment (ggilmore) [reply README.dev.md]: It can be, but that can be solved in a separate PR. I can file an issue for it.

File: README.dev.md (hunk 2 above)
Comment (ggilmore) [reply README.dev.md]: See https://github.com/sourcegraph/deploy-sourcegraph/pull/113#discussion_r226111067

File: README.dev.md (hunk 3 above)
Comment (ggilmore) [reply README.dev.md]: I decided to just link to install.md which has these quickstart instructions

File: docs/install.md [hunk 5]
```diff
@@ -46,7 +46,7 @@
    watch kubectl get pods -o wide
    ```
 
-1. Once the deployment completes, verify Sourcegraph is running by temporarily making the frontend port accessible:
+1. <a name="port-forward"></a>Once the deployment completes, verify Sourcegraph is running by temporarily making the frontend port accessible:
```
Comment (nicksnyder) [inline docs/install.md]: what does this do?

File: README.dev.md [hunk 6]
```diff
@@ -15,7 +15,65 @@
...
+  - Do the same smoke tests that we do for `sourcegraph/sourcegraph` (check to see that the new release works, check to see that the upgrade path works)
+    - Check to see that the latest `master` is working on a fresh cluster
+      1. Deploy the latest `master` to your new cluster by running through the quickstart steps in [docs/install.md](docs/install.md)
+         - You'll need to create a GCP Storage Class named `sourcegraph` with the same `zone` that you created your cluster in (see ["Configure a storage class"](./docs/configure.md#Configure-a-storage-class))
+         - In order to give yourself permissions to create roles on the cluster, run: `kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $YOUR_NAME@sourcegraph.com`
+      1. Use the instructions in [configure.md](./docs/configure.md) to add a repository, test that code intelligence is working on it, and do a couple searches
+    - Check the upgrade path from the previous release to the laster `master`
+      1. Tear down the cluster that you created above
```
Comment (nicksnyder) [inline README.dev.md]: a little bit of elaboration how would be nice

File: README.dev.md [hunk 7]
```diff
@@ -15,7 +15,65 @@
...
+      1. Checkout the latest `master`, deploy the newer images to the same cluster (without tearing it down in between) by running `./kubectl-apply-all.sh`, and check to see the smoke test passes after the upgrade process
+
+* The version numbers for [sourcegraph/deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph) largely follow [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph)'s version numbers (i.e. `deploy-sourcegraph@v2.11.2` uses [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph)'s `v2.11.2` image tags). Here is [sourcegraph/deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph)'s branching strategy:
+
+  - **If you are cutting a new minor version (e.g. `v2.12.0`)**:
+
+    - Make a branch named `v2.12` that stems from the current `master` branch and push it. `v2.12` will be used as the base for all `v2.12.x` images, and future changes to the `v2.12.x` series will be cherry-picked onto it and tagged from there.
```
Comment (nicksnyder) [inline README.dev.md]: branch names should not have a `v` prefix

File: docs/install.md (hunk 5 above)
Comment (ggilmore) [reply docs/install.md]: This was left over, I'll delete it. I was just manually creating an anchor so that I could link to it from another document.

**Comment by ggilmore** [comment]
@nicksnyder @beyang @attfarhan updated, ptal

File: README.dev.md [hunk 8]
```diff
@@ -1,39 +1,121 @@
...
+1.  Click on the `connect` button next to your cluster, it’ll give you a command to copy+paste in your terminal.
+1.  Run the command in your terminal. Once it finishes, run `kubectl config current-context`. It should tell you that it’s set up to talk to the cluster that you just created.
+
+### Do smoke tests for `master` branch
+
+1. Deploy the latest `master` to your new cluster by running through the quickstart steps in [docs/install.md](docs/install.md)
+   - You'll need to create a GCP Storage Class named `sourcegraph` with the same `zone` that you created your cluster in (see ["Configure a storage class"](docs/configure.md#Configure-a-storage-class))
+   - In order to give yourself permissions to create roles on the cluster, run: `kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $YOUR_NAME@sourcegraph.com`
+1. Use the instructions in [configure.md](docs/configure.md) to:
+   1. Ad a repository (e.g. [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph))
```
Comment (attfarhan) [inline README.dev.md]:
```suggestion
   1. Add a repository (e.g. [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph))
```

**Pull request merged**